compose/config/types.py

```diff
@@ -163,3 +163,9 @@ def parse(cls, volume_config):
     def repr(self):
         external = self.external + ':' if self.external else ''
         return '{ext}{v.internal}:{v.mode}'.format(ext=external, v=self)
+
+    @property
+    def is_named_volume(self):
+        return self.external and not (
+            self.external.startswith('.') or self.external.startswith('/')
```

Comment:
I believe `~` is the other path we consider as a host bind. We do a similar check in `split_path_mapping()`.

I think we could combine these and do something like:

```python
def is_host_bind(path):
    return path.startswith(('.', '~', os.sep))
```

Note: startswith can take a tuple: https://docs.python.org/2/library/stdtypes.html#str.startswith